frontend/src/components/sign_in/GoogleSignInButton.spec.tsx

```diff
@@ -24,27 +30,51 @@ it("calls the correct function on click", () => {
           onFailure={expectedFailure}
           scope={EXPECTED_SCOPES}
           text={EXPECTED_TEXT}
-          tokenConsumer={expectedConsumer} 
+          tokenConsumer={expectedConsumer}
         />
       </ServiceContext.Provider>
     </div>
   );
   TestUtils.Simulate.click(div.querySelector("button"));
-  expect(signInMock).toHaveBeenCalledWith(
-        EXPECTED_SCOPES, expectedConsumer, expectedFailure);
+  await expect(signInMockSuccessful).toHaveBeenCalledWith(EXPECTED_SCOPES);
+  expect(expectedConsumer).toHaveBeenCalledWith(TOKEN);
+});
+
+it("calls the correct function if failed", async () => {
+  const mockedAuthHandlerService = {
+    signIn: signInMockFailure,
+  };
+  const div = TestUtils.renderIntoDocument(
+    <div>
+      <ServiceContext.Provider
+        value={{authenticationHandlerService: mockedAuthHandlerService}}
+      >
+        <GoogleSignInButton
+          onFailure={expectedFailure}
+          scope={EXPECTED_SCOPES}
+          text={EXPECTED_TEXT}
+          tokenConsumer={expectedConsumer}
+        />
+      </ServiceContext.Provider>
+    </div>
+  );
+  TestUtils.Simulate.click(div.querySelector("button"));
+  await expect(signInMockSuccessful).toHaveBeenCalledWith(EXPECTED_SCOPES);
```

Comment:
nit: I don't think this await actually does anything. Also shouldn't it be signInMockFailure?

Reply:
Whoops, you're right about the signInMockFailure. The await is required however, without it the expected consumer will be called but after the next line runs.

Reply:
Hmm, interesting. I guess Jest matchers work differently from Jasmine matchers then, good to know.